docs/advanced.rst

```diff
@@ -199,19 +199,45 @@ development dependencies::
     py==1.4.34
     pytest==3.2.3
 
-Finally, if you wish to generate a requirements file with only the
+If you wish to generate a requirements file with only the
 development requirements you can do that too, using the ``--dev-only``
 flag::
 
     $ pipenv lock -r --dev-only
     py==1.4.34
     pytest==3.2.3
 
+Often, you would want to generate a requirements file based on your current
```

Comment:
Perhaps call this out as `Sometimes` instead of `Often` -- my concern here is that I've been realizing many folks are not taking full advantage of the dependency lock hash checking by just generating a requirements.txt and installing from that instead.   It would be good to make the documentation more clear about the benefits of using `pipenv install --deploy` or `pipenv sync` instead of generating requirements files because of the benefit of verification that the package version matches the same signature from when you locked/pinned to it.

Reply:
I completely agree. This should only be used for the use case of interfacing with tooling that only supports requirements.txt.